features/dd-sdk-android-webview/src/main/kotlin/com/datadog/android/webview/internal/replay/WebViewReplayEventConsumer.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.webview.internal.replay
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.api.context.DatadogContext
+import com.datadog.android.api.feature.Feature
+import com.datadog.android.api.feature.FeatureSdkCore
+import com.datadog.android.api.storage.DataWriter
+import com.datadog.android.webview.internal.WebViewEventConsumer
+import com.datadog.android.webview.internal.rum.TimestampOffsetProvider
+import com.datadog.android.webview.internal.rum.WebViewRumEventContextProvider
+import com.datadog.android.webview.internal.rum.domain.RumContext
+import com.google.gson.JsonObject
+import java.lang.IllegalStateException
+import java.lang.NumberFormatException
+import java.lang.UnsupportedOperationException
+
+internal class WebViewReplayEventConsumer(
+    private val sdkCore: FeatureSdkCore,
+    webViewId: Long,
+    internal val dataWriter: DataWriter<JsonObject>,
+    offsetProvider: TimestampOffsetProvider,
```

Comment:
minor: maybe it makes sense to simplify constructor by removing `webViewId` and `offsetProvider` here, removing the default value `webViewReplayEventMapper` and just creating it at the construction site? This way constructor will be simpler and shorter.